deploy-service/common/src/main/java/com/pinterest/deployservice/common/HTTPClient.java

```diff
@@ -47,27 +48,65 @@ private String generateUrlAndQuery(String url, Map<String, String> params) throw
         return sb.toString();
     }
 
+    // keep scrubbed/logged url generation separate for safety and easier extension
+    private String generateUrlAndScrubbedQuery(String url, Map<String, String> params) throws Exception {
+        if (params == null || params.isEmpty()) {
+            return url;
+        }
+        StringBuilder sb = new StringBuilder();
+        sb.append(url);
+        String prefix = "?";
+        String paramValue = "";
+        boolean isSensitiveValue = false;
+        for (Map.Entry<String, String> entry : params.entrySet()) {
+            isSensitiveValue = false;
+            paramValue = "";
+            sb.append(prefix);
+            prefix = "&";
+            for (String filteredQueryKeySubstring : filteredQueryKeySubstrings) {
```

Comment:
1. Put comment that this could be a perf. issue if list get too large or query gets too large
2. Only call query string scrubber for DEBUG level log ( INFO level prints only Base url, DEBUG prints Base + mask query params)